File: 392.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        p = 0
        q = 0
        while p < len(s) and q < len(t):
            if s[p] == t[q]:
                p += 1
            q += 1
        return p == len(s)
    
if __name__ == '__main__':
    s = input()
    t = input()
    sol = Solution()
    print(sol.isSubsequence(s, t))

File: cpp/392.cpp
#include<iostream>
#include<string>

using namespace std;

class Solution {
public:
    bool isSubsequence(string s, string t) {
        char *p = &s[0], *q = &t[0];
        while(p != &s[s.size()] && q != &t[t.size()]){
            if(*p == *q) p++;
            q++;
        }
        return p == &s[s.size()];
    }
};

int main() {
    Solution sol;
    string s, t;
    cin >> s >> t;
    cout << sol.isSubsequence(s, t) << endl;
    return 0;
}